packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -222,6 +224,21 @@ class DefaultAWSAccount {
 
   private async lookupDefaultAccount(): Promise<string | undefined> {
     try {
+      // There just is *NO* way to do AssumeRole credentials as long as AWS_SDK_LOAD_CONFIG is not set. The SDK
+      // crash if the file does not exist though. So set the environment variable if we can find that file.
+      await setConfigVariable();
+
+      // support these
+      // This might try to load a profile which as AssumeRole credentials. This will normally work for the JS SDK
+      // if AWS_PROFILE/AWS_SDK_LOAD_CONFIG are set, but we don't set those (we work directly from the API).
+      //
+      // Since there is no way to pass a { region } argument to the STS client used for AssumeRole credentials
+      // (https://github.com/aws/aws-sdk-js/issues/2377), we must now configure the region globally so that it will
+      // be picked up there.
+      // AWS.config.update({
```

Comment:
Is this not true anymore? I am confused...

Reply:
Ah, you're right. This wasn't sufficient. We *MUST* load the config, and we *MUST* set the env var to do so. Will remove.